pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -107,6 +111,17 @@ func NewEndpointBuilder(clusterName string, proxy *model.Proxy, push *model.Push
 	return b
 }
 
+// NewTestEndpointBuilder allows setting some fields directly without needing push context.
+// For now, this is for tests. We can expand this to be a fully explicit constructor for cases
+// like CDS where we already have the service and destination rule.
+func NewTestEndpointBuilder(clusterName string, service *model.Service, dr *model.ConsolidatedDestRule) EndpointBuilder {
```

Comment:
nit: move to test files

Reply:
I would like to but:
1. Some of the tests are `endpoints_test` instead of `endpoints` package to avoid some cyclic imports.
2. Need access to unexported fields.

I think some future PRs will find a full constructor useful, we can consolidate then.